src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -197,235 +169,226 @@ public static Object returnPDFS(Map<String, Map<String, Object>> participantESDa
+                for (int i = 0; i < oncHistoryIDs.size(); i++) {
+                    OncHistoryDetail oncHistoryDetail = OncHistoryDetail.getOncHistoryDetail(oncHistoryIDs.get(i), ddpInstance.getName());
+                    // facility information is the same in all of the requests so only need to be set ones!
+                    if (i == 0) {
+                        valueMap.put(RequestPDFProcessor.FIELD_CONFIRMED_INSTITUTION_NAME, oncHistoryDetail.getFacility());
+                        valueMap.put(RequestPDFProcessor.FIELD_CONFIRMED_PHONE, oncHistoryDetail.getFPhone());
+                        valueMap.put(RequestPDFProcessor.FIELD_CONFIRMED_FAX, oncHistoryDetail.getFFax());
                     }
+                    valueMap.put(RequestPDFProcessor.FIELD_DATE_PX + i, oncHistoryDetail.getDatePX());
+                    valueMap.put(RequestPDFProcessor.FIELD_TYPE_LOCATION + i, oncHistoryDetail.getTypePX());
+                    valueMap.put(RequestPDFProcessor.FIELD_ACCESSION_NUMBER + i, oncHistoryDetail.getAccessionNumber());
+                    counter = i;
                 }
             }
-            catch (SQLException ex) {
-                dbVals.resultException = ex;
+            valueMap.put(RequestPDFProcessor.BLOCK_COUNTER, counter + 1);
+
+            stream = processor.generateStream(valueMap);
+            stream.mark(0);
+            return IOUtils.toByteArray(stream);
+        }
+        catch (IOException e) {
+            throw new RuntimeException("Couldn't get pdf for participant " + ddpParticipantId + " of ddpInstance " + ddpInstance.getName(), e);
+        }
+        finally {
+            try {
+                if (stream != null) {
+                    stream.close();
+                }
             }
-            return dbVals;
-        });
+            catch (IOException e) {
+                throw new RuntimeException("Error closing stream", e);
+            }
+        }
+    }
 
-        if (results.resultException != null) {
-            throw new RuntimeException("Error getting instance data for participant " + ddpParticipantId, results.resultException);
+    private byte[] getIRBLetter(@NonNull DDPInstance ddpInstance) {
+        String groupId = DDPInstance.getDDPGroupId(ddpInstance.getName());
+        return PDFProcessor.getTemplateFromGoogleBucket(groupId + "_IRB_Letter.pdf");
+    }
+
+    /**
+     * get pdf
+     * for the given participant and pdfType
+     *
+     * @throws Exception
+     */
+    public void getPDFs(@NonNull Response response, @NonNull String ddpParticipantId, String pdfType, String medicalRecordId,
+                        List<String> oncHistoryIDs, List<PDF> pdfs, String realm, String requestBody, Integer userId) {
+        DDPInstance ddpInstance = DDPInstance.getDDPInstance(realm);
+        if (ddpInstance != null && StringUtils.isNotBlank(ddpParticipantId)) {
+            byte[] pdfBytes = null;
+            String fileName = "";
+            if (COVER.equals(pdfType)) {
```

Comment:
with PDF abstraction you would introduce abstract factory design pattern which would help to determine which PDF object needed to be created for PDF processing